src/app/labeling/qgsmaptoolmovelabel.cpp

```diff
@@ -293,8 +293,30 @@ void QgsMapToolMoveLabel::cadCanvasPressEvent( QgsMapMouseEvent *e )
           vlayer->beginEditCommand( tr( "Moved label" ) + QStringLiteral( " '%1'" ).arg( currentLabelText( 24 ) ) );
         else
           vlayer->beginEditCommand( tr( "Moved callout" ) );
-        bool success = vlayer->changeAttributeValue( featureId, xCol, xPosNew );
-        success = vlayer->changeAttributeValue( featureId, yCol, yPosNew ) && success;
+
+        // Try to convert to the destination field type
+        QVariant xNewPos( xPosNew );
+        QVariant yNewPos( yPosNew );
+
+        if ( xCol < vlayer->fields().count() )
+        {
+          if ( ! xNewPos.convert( vlayer->fields().at( xCol ).type() ) )
```

Comment:
Shouldn't this use QgsField::convertCompatible?

Reply:
done